specs/AnimatedVisualPlayer Spec.md

```diff
@@ -0,0 +1,145 @@
+AnimatedVisualPlayer Spec
+===
+
+# Background
+
+Today, developers can create animated visuals to use in their UI by using the
+[AnimatedVisualPlayer](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.AnimatedVisualPlayer)
+control.
+`AnimatedVisualPlayer` can play any
+[IAnimatedVisual](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisual)
+implementation,
+but the typical way to get an `IAnimatedVisual` implementation is to use
+the LottieGen tool to generate code from a JSON file that was created from Adobe AfterEffects.
+
+[More information on Lottie and AnimatedVisualPlayer](https://docs.microsoft.com/en-us/windows/communitytoolkit/animations/lottie)
+
+There are some performance issues with the current implementation of LottieGen and AnimatedVisualPlayer: 
+1)	Complex Lottie animations can instantiate a lot of composition animations 
+which may impact performance and overall battery life. 
+2)	Lottie files that have a lot of complex animations even when they are paused
+have some cost to performance. 
+The Idle (paused) Lottie animations use the same amount of CPU resources as
+when they are playing the actual animation.
+
+The new APIs here allow `AnimatedVisualPlayer`,
+[AnimatedIcon](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.AnimatedIcon),
+and the interface generated by LottieGen to be updated to help reduce the number of composition animations created
+when the animation is idle or paused to help improve performance when a developer uses
+Lottie Animations with `AnimatedVisualPlayer`.
+
+This spec will focus on changes for LottieGen and `AnimatedVisualPlayer`. 
+
+This will require 3 changes: 
+1. Create a new `IAnimatedVisual2` interface which adds to
+[IAnimatedVisual](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisual)
+the ability to destroy and instantiate animations.
+2. Create a new `IAnimatedVisualSource3` interface that extends
+[IAnimatedVisualSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisualSource)
+with the ability to return the new `IAnimatedVisual2`
+3.	Add a `CacheMode` property to `AnimatedVisualPlayer` that will control
+when animations should be destroyed and when they should be instantiated.
+This will be implemented using the previous new interfaces.
+
+# API Pages
+
+## AnimatedVisualPlayer.CacheMode property
+
+Gets or sets a value that determines how animations are cached. Defaults to 'Always'.
+
+The two modes for the property will specify the behavior the AnimatedVisualPlayer will have
+when the player is idle (when `PlayAsync` is not active).
+
+|Mode| Behavior|
+|---- | -------|
+|Always| `AnimatedVisualPlayer` will pre-create animations even before `PlayAsync()` is called,
+and not destroy any when player is idle. |
+|None | `AnimatedVisualPlayer` will not create animations until `PlayAsync()` is called,
+and will destroy them when it completes. |
```

Comment:
Table rows cannot be split across multiple lines. See markdown output: The second line is treated as a new table row.